devops/providers/azure-devops/templates/infrastructure/README.md

```diff
@@ -0,0 +1,204 @@
+
+### Staging
+
+The staging environment is a 1:1 replica of the production environment.
+
+Cobalt automated tests should be tagged to application environment targets to avoid automated tests that happen to mutate data in production.
+
+This Azure Devops stage does not support parallel releases due to the dedicated resources.
+
+#### Release Gate
+
+There's a gate prior to applying the execution plan onto the Staging environment. The technical operator should review the terraform execution plan prior to approving the gate request.
+
+### Production
+
+The production stage release is auto triggered upon feature branch merges into master. This stage will apply the desired terraform changes within your production environment.
+
+One git scm flow worth considering is the [GitHub flow](https://guides.github.com/introduction/flow/). This model promotes short feedback loops for feature branch pull requests. This means work branches (‘work’ could mean a new feature or a bug fix – there is no distinction) starts from the production code (master) and are short lived.
+
+Our pipeline can support any scm git workflow(gitflow, etc).
+
+The master branch is a record of known good production code. Feature branch(s) should only be merged into master once: 1) code review's are approved 2) all CD pre-prod releases are complete. These can be validated as part of your master branch policy.
+
+This Azure Devops stage does not support parallel releases due to the dedicated resources.
+
+## Pipeline setup
+
+As discussed earlier, there's one git repository for each infrastructure template, as technical operators are code owners of cloud resource manifests.
+
+### Add Baseline Azure Build Pipeline
+
+- Within your azure devops project, create a new pipeline
+
+![image](https://user-images.githubusercontent.com/7635865/56069362-549b4080-5d48-11e9-97b9-02cb01cc5b35.png)
+
+- Select GitHub as your source
+
+![image](https://user-images.githubusercontent.com/7635865/56069729-05eea600-5d4a-11e9-8aa8-002feb8519a0.png)
+
+- After selecting your service connection, provide the location of your target repository.
+
+![image](https://user-images.githubusercontent.com/7635865/56069808-5fef6b80-5d4a-11e9-9d5d-d4a7fb372a41.png)
+
+- Point the build definition to the repository's target yaml pipeline location.
+
+![image](https://user-images.githubusercontent.com/7635865/56069873-a5ac3400-5d4a-11e9-80e0-fe2e90b5639b.png)
+
+![image](https://user-images.githubusercontent.com/7635865/56069976-2c611100-5d4b-11e9-9bc0-b4dad6d1cd9c.png)
+
+### Application Dev Team Pipeline Setup
+
+There should be one Azure Unified pipeline for each individual application. This can be achieved by cloning the baseline build definition for each application per azure devops instance.
+
+![image](https://user-images.githubusercontent.com/7635865/60265046-44582380-98aa-11e9-9644-4d04cb561b79.png)
+
+## Template Variables
```

Comment:
Can the `Template Variables` needed be defined more generically as the same ones needed by the template being deployed?